### PR TITLE
fix(evpn, unit test):fix evpn and unit test

### PR DESCRIPTION
**Pull request by atulpatel261194**

### CODE REVIEW

File: cmd/network/evpn-logical-brige.go [hunk 1]
```diff
@@ -59,9 +68,6 @@ func CreateLogicalBridge() *cobra.Command {
 	if err := cmd.MarkFlagRequired("vlan-id"); err != nil {
 		log.Fatalf("Error marking flag as required: %v", err)
 	}
-	if err := cmd.MarkFlagRequired("vni"); err != nil {
```
Comment (glimchb) [inline cmd/network/evpn-logical-brige.go:62]: can you use `MarkFlagsMutuallyExclusive` or `MarkFlagsRequiredTogether` ?

File: cmd/network/evpn-logical-brige.go [hunk 2]
```diff
@@ -28,14 +28,23 @@ func CreateLogicalBridge() *cobra.Command {
...
+			var vniparam *uint32
 			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 			evpnClient, err := network.NewLogicalBridge(addr)
 			if err != nil {
 				log.Fatalf("could create gRPC client: %v", err)
 			}
 			defer cancel()
 
-			lb, err := evpnClient.CreateLogicalBridge(ctx, name, vlanID, vni, vtep)
+			if (vni == 0 && vtep != "") || (vtep == "" && vni != 0) {
```
Comment (glimchb) [inline cmd/network/evpn-logical-brige.go:39]: can you use `MarkFlagsMutuallyExclusive` or `MarkFlagsRequiredTogether` ?

File: cmd/network/evpn-vrf.go [hunk 3]
```diff
@@ -49,10 +52,6 @@ func CreateVRF() *cobra.Command {
 	cmd.Flags().StringVar(&vtep, "vtep", "", "VTEP IP address")
 	cmd.Flags().StringVar(&addr, "addr", "localhost:50151", "address of OPI gRPC server")
 
-	if err := cmd.MarkFlagRequired("vni"); err != nil {
```
Comment (glimchb) [inline cmd/network/evpn-vrf.go:52]: can you use `MarkFlagsMutuallyExclusive` or `MarkFlagsRequiredTogether` ?

File: cmd/network/evpn-vrf.go [hunk 4]
```diff
@@ -27,14 +27,17 @@ func CreateVRF() *cobra.Command {
...
+			var vniparam *uint32
 			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 			evpnClient, err := network.NewVRF(addr)
 			if err != nil {
 				log.Fatalf("could not create gRPC client: %v", err)
 			}
 			defer cancel()
-
-			vrf, err := evpnClient.CreateVrf(ctx, name, vni, loopback, vtep)
+			if vni != 0 {
```
Comment (glimchb) [inline cmd/network/evpn-vrf.go:37]: can you use `MarkFlagsMutuallyExclusive` or `MarkFlagsRequiredTogether` ?

**Review by glimchb — CHANGES_REQUESTED** [review]
small suggestion inside

File: cmd/network/evpn-vrf.go (hunk 3 above)
Comment (atulpatel261194) [reply cmd/network/evpn-vrf.go:52]: I believe its not possible as LoopbackIpPrefix in only mandatory flag, vni and vtep are optional.

File: cmd/network/evpn-vrf.go (hunk 4 above)
Comment (atulpatel261194) [reply cmd/network/evpn-vrf.go:37]: I believe its not possible as LoopbackIpPrefix is only mandatory flag, vni and vtep are optional.

**Review by glimchb — APPROVED** [review]
lgtm

File: network/logical_bridge.go [hunk 5]
```diff
@@ -24,17 +27,21 @@ func (c evpnClientImpl) CreateLogicalBridge(ctx context.Context, name string, vl
...
 	client := c.getEvpnLogicalBridgeClient(conn)
-	ipVtep, err := parseIPAndPrefix(vtepIP)
-	if err != nil {
-		log.Printf("parseIPAndPrefix: error creating vrf: %s\n", err)
-		return nil, err
+	if vni != nil {
+		if vtepIP != "" {
+			ipVtep, err = parseIPAndPrefix(vtepIP)
+			if err != nil {
+				log.Printf("parseIPAndPrefix: error creating vrf: %s\n", err)
```
Comment (artek-koltun) [inline network/logical_bridge.go:34]: parseIPAndPrefix: error creating **Logical Bridge**: %s\n ?

File: network/logical_bridge.go [hunk 6]
```diff
@@ -24,17 +27,21 @@ func (c evpnClientImpl) CreateLogicalBridge(ctx context.Context, name string, vl
 	defer closer()
 
 	client := c.getEvpnLogicalBridgeClient(conn)
-	ipVtep, err := parseIPAndPrefix(vtepIP)
-	if err != nil {
-		log.Printf("parseIPAndPrefix: error creating vrf: %s\n", err)
-		return nil, err
+	if vni != nil {
+		if vtepIP != "" {
```
Comment (artek-koltun) [inline network/logical_bridge.go:31]: we can re-write it to have less nesting levels
```go
if vni != nil && vtepIP != "" {
```
`vni` and `vtepIP` are required together, should we report an error if any of them missed?

File: network/vrf.go [hunk 7]
```diff
@@ -29,16 +31,20 @@ func (c evpnClientImpl) CreateVrf(ctx context.Context, name string, vni uint32,
 		log.Printf("parseIPAndPrefix: error creating vrf: %s\n", err)
 		return nil, err
 	}
-	ipVtep, err := parseIPAndPrefix(vtep)
-	if err != nil {
-		log.Printf("parseIPAndPrefix: error creating vrf: %s\n", err)
-		return nil, err
+	if vni != nil {
+		if vtep != "" {
```
Comment (artek-koltun) [inline network/vrf.go:35]: the same as in `CreateLogicalBridge`

**Review by artek-koltun — APPROVED** [review]
LGTM

**Pull request merged**